server/services/DeckService.js

```diff
@@ -37,13 +37,23 @@ class DeckService {
         return this.decks.find({ standaloneDeckId: { $exists: true } }, { sort: { lastUpdated: -1 } });
     }
 
-    create(deck) {
+    async create(deck) {
+        //if the eventId is set on a new deck, check if the user already has a deck with the same eventId
+        if(deck.eventId) {
+            //if a deck for the event already exists, do not create the new deck
+            if(this.userAlreadyHasDeckForEvent(deck.userName, deck.eventId)) {
+                return () => Promise.resolve();
```

Comment:
Just throw an error instead of resolving to nothing

Reply:
fixed